### PR TITLE
cpp-qt-client: Fix cppcheck warnings

### PR DESCRIPTION
**Pull request by MartinDelille**
Here is a list of improvement to fix most of the cppcheck and clazy warning by a generated code:
- Remove unsused variable
- Don't create temporary item when iterating
- Pass QString parameter by const reference
- Pass a context object as 3rd connect parameter (https://github.com/KDE/clazy/blob/master/docs/checks/README-connect-3arg-lambda.md)

The second commit improve the coding style.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (5.3.0), `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@ravinikam @stkrwork @etherealjoy @muttleyxd

### CODE REVIEW

File: modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache [hunk 1]
```diff
@@ -11,19 +11,21 @@ namespace {{this}} {
...
 }
 
-void {{classname}}::initializeServerConfigs(){
+void {{classname}}::initializeServerConfigs() {
     //Default server
     QList<{{prefix}}ServerConfiguration> defaultConf = QList<{{prefix}}ServerConfiguration>();
     //varying endpoint server
+    {{#servers}}
     QList<{{prefix}}ServerConfiguration> serverConf = QList<{{prefix}}ServerConfiguration>();
+{{/servers}}
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/cpp-qt-client/api-body.mustache:28]: Suggestion: add 4-space at the beginning.

**Comment by wing328** [comment]
LGTM but I've not tested it locally

**Comment by wing328** [comment]
Looks like this breaks the build:
```
I/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client -I/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/build/client/client_autogen/include -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++-64 -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork  -fPIC -Wall -Wno-unused-variable -fPIC -Wall -Wno-unused-variable   -fPIC -std=c++14 -o CMakeFiles/client.dir/PFXPetApi.cpp.o -c /home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp: In member function â€˜void test_namespace::PFXPetApi::setNewServerForAllOperations(const QUrl&, const QString&, const QMap<QString, test_namespace::PFXServerVariable>&)â€™:
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:129:38: error: â€˜class QMap<QString, int>â€™ has no member named â€˜keyBeginâ€™
     for (auto keyIt = _serverIndices.keyBegin(); keyIt != _serverIndices.keyEnd(); keyIt++) {
                                      ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:129:74: error: â€˜class QMap<QString, int>â€™ has no member named â€˜keyEndâ€™
     for (auto keyIt = _serverIndices.keyBegin(); keyIt != _serverIndices.keyEnd(); keyIt++) {
                                                                          ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp: In member function â€˜void test_namespace::PFXPetApi::addPet(const test_namespace::PFXPet&)â€™:
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:236:44: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueBeginâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                            ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:236:91: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueEndâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                                                                           ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp: In member function â€˜void test_namespace::PFXPetApi::deletePet(const qint64&, const test_namespace::OptionalParam<QString>&)â€™:
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:298:44: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueBeginâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                            ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:298:91: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueEndâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                                                                           ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp: In member function â€˜void test_namespace::PFXPetApi::findPetsByStatus(const QList<QString>&)â€™:
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:426:44: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueBeginâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                            ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:426:91: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueEndâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                                                                           ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp: In member function â€˜void test_namespace::PFXPetApi::findPetsByTags(const QList<QString>&)â€™:
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:564:44: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueBeginâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                            ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:564:91: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueEndâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                                                                           ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp: In member function â€˜void test_namespace::PFXPetApi::getPetById(const qint64&)â€™:
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:634:44: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueBeginâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                            ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:634:91: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueEndâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                                                                           ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp: In member function â€˜void test_namespace::PFXPetApi::updatePet(const test_namespace::PFXPet&)â€™:
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:681:44: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueBeginâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                            ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:681:91: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueEndâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                                                                           ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp: In member function â€˜void test_namespace::PFXPetApi::updatePetWithForm(const qint64&, const test_namespace::OptionalParam<QString>&, const test_namespace::OptionalParam<QString>&)â€™:
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:745:44: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueBeginâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                            ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:745:91: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueEndâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                                                                           ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp: In member function â€˜void test_namespace::PFXPetApi::uploadFile(const qint64&, const test_namespace::OptionalParam<QString>&, const test_namespace::OptionalParam<test_namespace::PFXHttpFileElement>&)â€™:
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:809:44: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueBeginâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                            ^
/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/client/PFXPetApi.cpp:809:91: error: â€˜class QMap<QString, QString>â€™ has no member named â€˜keyValueEndâ€™
     for (auto keyValueIt = _defaultHeaders.keyValueBegin(); keyValueIt != _defaultHeaders.keyValueEnd(); keyValueIt++) {
                                                                                           ^
client/CMakeFiles/client.dir/build.make:143: recipe for target 'client/CMakeFiles/client.dir/PFXPetApi.cpp.o' failed
make[2]: Leaving directory '/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/build'
make[2]: *** [client/CMakeFiles/client.dir/PFXPetApi.cpp.o] Error 1
make[1]: *** [client/CMakeFiles/client.dir/all] Error 2
CMakeFiles/Makefile2:164: recipe for target 'client/CMakeFiles/client.dir/all' failed
make[1]: Leaving directory '/home/travis/build/OpenAPITools/openapi-generator/samples/client/petstore/cpp-qt/build'
Makefile:132: recipe for target 'all' failed
make: *** [all] Error 2
```

Can you please take a look when you've time?

Travis build is run for master only due to free credits running low:

https://app.travis-ci.com/github/OpenAPITools/openapi-generator/builds/237210677

I'll comment out the job for the cpp test for the time being.

**Comment by MartinDelille** [comment]
Ok I will have a look and revert the specific change if needed. Which Qt version runs the CI ?

**Pull request merged**